cmd/collector/app/builder/builder_flags.go

```diff
@@ -32,6 +32,7 @@ const (
 	collectorGRPCTLS              = "collector.grpc.tls"
 	collectorGRPCCert             = "collector.grpc.tls.cert"
 	collectorGRPCKey              = "collector.grpc.tls.key"
+	collectorGRPCClientCA         = "collector.grpc.tls.client.ca"
```

Comment:
This is slightly inconsistent with the flags in the agent, where it's called just `tls.ca`. Is there a logical reason to have that difference?

If not, it would be nice to refactor this whole TLS flags into a shared `pkg/tls` component, that would include defining & parsing flags and instantiating the objects. NB: it's ok if you'd rather not do that, we can do it in a separate PR, I just want to get your thoughts on whether it's a feasible approach. We seem to have quite a lot of these already, e.g. in storage:
```shell
$ grep -r .tls.cert plugin
plugin/storage/cassandra/options.go:	suffixCert                 = ".tls.cert"
plugin/storage/es/options.go:	suffixCert              = ".tls.cert"
```

Even in this PR we're repeating code to load cert/key/etc twice.

Reply:
There are two CAs involved, for two different purposes. One verifies hosts connected to by outbound hosts, the other checks certs for incoming clients. In this current case, each service only needs to do one of those, so only take one CA (though use it for different purposes). I'd suggest keeping the names distinct in the event that you extend the approach in this PR to other components.